src/screens/ENSConfirmRegisterSheet.js

```diff
@@ -329,70 +265,94 @@ export default function ENSConfirmRegisterSheet() {
+          isValidGas={
+            registrationCostsData?.isValidGas &&
+            Boolean(registrationCostsData?.stepGasLimit)
+          }
+          label={lang.t('profiles.confirm.confirm_renew')}
+          testID={step}
+        />
+      ),
       [REGISTRATION_STEPS.EDIT]: (
         <TransactionActionRow
           accentColor={accentColor}
           action={() => action(goToProfileScreen)}
-          isSufficientGas={isSufficientGasForStep}
-          isValidGas={isValidGas && Boolean(gasLimit)}
+          isSufficientGas={registrationCostsData?.isSufficientGasForStep}
+          isValidGas={
+            registrationCostsData?.isValidGas &&
+            Boolean(registrationCostsData?.stepGasLimit)
+          }
           label={lang.t('profiles.confirm.confirm_update')}
           testID={step}
         />
       ),
+      [REGISTRATION_STEPS.SET_NAME]: (
+        <TransactionActionRow
+          accentColor={accentColor}
+          action={() => action(goToProfileScreen)}
+          isSufficientGas={registrationCostsData?.isSufficientGasForStep}
+          isValidGas={
+            registrationCostsData?.isValidGas &&
+            Boolean(registrationCostsData?.stepGasLimit)
+          }
+          label={lang.t('profiles.confirm.confirm_set_name')}
+          testID={step}
+        />
+      ),
       [REGISTRATION_STEPS.WAIT_COMMIT_CONFIRMATION]: null,
       [REGISTRATION_STEPS.WAIT_ENS_COMMITMENT]: null,
     }),
     [
       accentColor,
       action,
-      gasLimit,
-      goToProfileScreen,
-      isSufficientGasForStep,
-      isValidGas,
       registrationCostsData?.isSufficientGasForRegistration,
+      registrationCostsData?.isSufficientGasForStep,
+      registrationCostsData?.isValidGas,
+      registrationCostsData?.stepGasLimit,
       step,
+      goToProfileScreen,
     ]
   );
 
-  // Update gas limit
   useEffect(() => {
-    if (
-      stepGasLimit &&
-      !isEmpty(gasFeeParamsBySpeed) &&
-      gasLimit !== stepGasLimit
-    ) {
-      updateTxFee(stepGasLimit);
-      setGasLimit(stepGasLimit);
-    }
-  }, [gasFeeParamsBySpeed, gasLimit, stepGasLimit, updateTxFee]);
+    registrationCostsData?.hasReverseRecord !== undefined &&
+      setSendReverseRecord(!registrationCostsData?.hasReverseRecord);
+  }, [registrationCostsData?.hasReverseRecord]);
 
-  useEffect(() => {
-    if (
-      step === REGISTRATION_STEPS.COMMIT ||
-      step === REGISTRATION_STEPS.REGISTER ||
-      step === REGISTRATION_STEPS.EDIT
-    )
-      startPollingGasFees();
-  }, [startPollingGasFees, step]);
-
-  useEffect(() => {
-    // if reverse record is set, we don't want to send the reverse record tx by default
-    const getReverseRecord = async () => {
-      const reverseRecord = await fetchReverseRecord(accountAddress);
-      if (reverseRecord) setSendReverseRecord(false);
-    };
-    getReverseRecord();
-  }, [accountAddress]);
+  useFocusEffect(
+    useCallback(() => {
```

Comment:
this useCallback doesn't seem necessary

Reply:
`useFocusEffect` is not memoized by default, and I believe `blurFields` is quite expensive, so I don't want to be calling it on every render.